tapis-project/tapis-ui
# task/TUI-265 -- file browsing

## Overview:

Allow browsing storage systems for files for source URLs

## Related Github Issues:

- [TUI-265](https://github.com/tapis-project/tapis-ui/issues/265)

## Summary of Changes:

- Refactor FileExplorer from tapis-app to tapis-ui
- FileSelectModal component
- useModal hook for GenericModal
- FormikTapisFile and FormikTapisFileInput components
- FileInputs and FileInputArrays steps now allow browsing for sourceUrls
- FileInputArrays allows bulk addition of files via browsing
- Fix: Navigating with FileExplorer was causing onSelect callback to fire

## Testing Steps:

1. Try the Browse button in FileInputs with existing source URLs and urls that are not `tapis://` URIs
2. Try the `+ Browse for files` button in FileInputArrays
3. Test pre-existing Files browser (should still be able to navigate, select and perform operations on selected files)

## UI Photos:

![image](https://user-images.githubusercontent.com/17181582/160193595-39857c23-6144-4098-ba09-e17718c1ef27.png)

![image](https://user-images.githubusercontent.com/17181582/160194073-4dbeb699-2e93-456f-934a-0bc3ca8d9080.png)

## Notes:

Working on unit tests
